cmd/ndt7-client/main.go

```diff
@@ -183,10 +191,25 @@ func main() {
 		InsecureSkipVerify: *flagNoVerify,
 	}
 	r.client.FQDN = *flagHostname
-	if *flagBatch {
-		r.emitter = emitter.NewBatch()
+
+	var e emitter.Emitter
+
+	// If -batch, force -format=json.
+	if *flagBatch || *flagFormat == "json" {
+		e = emitter.NewJSON()
 	} else {
-		r.emitter = emitter.NewInteractive()
+		e = emitter.NewHumanReadable()
 	}
-	osExit(r.runDownload(ctx) + r.runUpload(ctx))
+	if *flagQuiet {
+		e = emitter.NewQuiet(e)
+	}
+	r.emitter = e
+
+	code := r.runDownload(ctx) + r.runUpload(ctx)
+	if code != 0 {
+		osExit(code)
+	}
+
+	s := internal.NewSummary(r.client.FQDN, r.client.Results())
+	r.emitter.OnSummary(s)
```

Comment:
I agree that it's tricky to decide whether to show a summary if there was an error